evap/evaluation/models.py

```diff
@@ -284,7 +287,11 @@ def can_user_see_results(self, user):
         if user.is_staff:
             return True
         if self.state == 'published':
-            return self.can_publish_grades or self.is_user_contributor_or_delegate(user)
+            if self.is_user_contributor_or_delegate(user):
+                return True
+            if not self.can_publish_grades:
+                return False
+            return not self.is_private or user in self.participants.all()
```

Comment:
i will change it.
i see what you mean and it's okay for me and better than before but still not optimal. shouldn't restricting be actually the default case, so that we only explicitly allow some cases and restrict everything else? now there's a `return True` that might do unexpected stuff if we should ever have unhandled cases in there.
but i don't mind having at that way because i don't have a better solution in mind right now.